### ISSUE TITLE
RFC1123 dates not properly recognized by akka-http if single-digit-day

### ISSUE THREAD

**Issue by neurolabs** [akka/akka-http#1110]
AFAIUnderstand, this date is RFC1123-HTTP-date-compatible: "Tue, 9 May 2017 18:49:57 GMT". The day can be either one or two digits. This format is generated by:

`ZonedDateTime.of(1986, 4, 8, 12, 30, 0, 0, ZoneId.of("GMT")).format(DateTimeFormatter.RFC_1123_DATE_TIME)`

But this does not parse in Akka HTTP, due to akka.http.impl.model.parser.CommonRules#day requiring the day field to always consist of two digits.

This breaks caching for me, since an 'If-Modified-Since' header isn't properly interpreted and the server delivers a 200 response instead of a 304.

**Comment by jrudolph** [comment]
Thanks for the report @neurolabs (and welcome here :) ).

Indeed, [RFC 1123 Section 5.2.14](https://tools.ietf.org/html/rfc1123#section-5.2.14) gives this as grammar for the date:

```
The syntax for the date is hereby changed to:

            date = 1*2DIGIT month 2*4DIGIT
```

So, both one or two digit days should be allowed. Should be simple to change in the grammar.

**Comment by neurolabs** [comment]
Oh, good point @jrudolph, years currently are required to consist of 4 digits as well, should also allow two digits.

**Comment by jrudolph** [comment]
Fixed in #1124.
